Room personal name (#3015)
matrix-org/matrix-doc: proposals/3015-room-personal-name.md

# MSC3015: Room personal name

> Very often users want to personally rename rooms to see it in list like they wants, especially for DM rooms. But room name is shared thing, so if they rename it for yourself, all other members will see this rename too. 

Reviewer: please line wrap at about 120 characters for legibility.

Author: Thanks for suggestion, done!

# MSC3015: Room personal name

Very often users want to personally rename rooms to see it in list like they wants, especially for DM rooms. But room
name is shared thing, so if they rename it for yourself, all other members will see this rename too. 

Examples of problem: 

> 1. Very often other people want to rename DM rooms with me and can do this, because we both are admins. So he set the

Reviewer 2: ```suggestion
1. Very often other people want to rename DM rooms with me and can do this, because we both are admins. So they set the
```

>    alternative name of that DM room (eg `Korepov Alexey` instead of `Alexey Murz Korepov`) - this works well on his

Reviewer 2: ```suggestion
   alternative name of that DM room (eg `Korepov Alexey` instead of `Alexey Murz Korepov`) - this works well on their
```

   side. But, as result on my side, I see that room in my rooms list with my name, instead of remote user name.

> 2. The user have two DM rooms with different Matrix users, but both have name "Alice" without avatar. As result he see

Reviewer 2: ```suggestion
2. The user has two DM rooms with different Matrix users, but both have name "Alice" without avatar. As result they see
```

>    two identical rooms in list with same name and can understand which Alice is where only via reading recent messages.

Reviewer 2: ```suggestion
   two identical rooms in the list with same name and it is unclear which room is which.
```

>    And he can solve this problem via renaming rooms, that give the result from item 1, descibed above.

Reviewer 2: ```suggestion
   If attempting to solve this problem via renaming rooms, the problem described in 1 occurs.
```

3. This problem often happens for rooms from bridged networks, when we talk with same person via different networks, and
   want to mark each room personally. This can be solved via adding suffixes with remote network name to room name on
   Bridge side, but people want to change other parts of room name too. For example, one person can have different names
   in each network (eg "Alexey Korepov" on VK, "Korepov Alexey" on Skype, "Alex" on WhatsApp, "Murz" on Telegram), and I

>    want to see all rooms with him as similar names.

Reviewer 2: ```suggestion
   want to see all the rooms with similar names.
```

> Most of other modern messengers (Telegram, Skype, Viber, WhatsApp) already have this feature, but only for DM rooms via

Reviewer 2: ```suggestion
Most other modern messengers (Telegram, Skype, Viber, WhatsApp) already have this feature, but only for DM rooms via
```

> reusing smartphone's addressbook to store personal names of contacts. And moving from that messengers to Matrix confuses

Reviewer 2: ```suggestion
reusing smartphone's addressbook to store personal names of contacts. Moving from these messengers to Matrix is confusing
```

> the people, because they can't rename personal chats in his own list like before.

Reviewer 2: ```suggestion
because they can't add personal names to chats like before.
```

# Proposal

For solve this problem I propose to use [room's
account_data](https://matrix.org/docs/spec/client_server/r0.6.0#put-matrix-client-r0-user-userid-rooms-roomid-account-data-type)
item with key `m.room_name_personal` to store custom name of any room for each user individually, with filling personal
name in content with `room_name_personal` type:

```json
{
  "room_name_personal": "Alice Liddell"
}
```

> By default this item is absent. It is added only when user make the personal renaming of room, and cleared if user

Reviewer 2: ```suggestion
By default this item is absent. It is added only when user sets a personal name for the room, and cleared if user
```

> remove the personal name for room (or make it empty).

Reviewer 2: ```suggestion
removes the personal name for the room (or make it empty).
```

> remove the personal name for room (or make it empty).

Reviewer 2: I don't think we should support "making it empty" as a feature. This can be a client UX decision that an empty room name is either not allowed or translates to clearing the personal name on the matrix side.

> Regarding to spec, the account data's key can't be deleted, so if user wants to clean the personal name or "Reset to

Reviewer 2: ```suggestion
According to the spec, account data keys can't be deleted, so if user wants to clean the personal name or "Reset to
```

default", the value of the `m.room_name_personal` key should become empty (`{}`).

# Client support

## Displaying:

> When client displays the room in list, it should lookup the `m.room_name_personal` key, if it exists and have not empty

Reviewer 2: ```suggestion
When client displays the room in list, it should lookup the `m.room_name_personal` key, if it exists and has a `room_name_personal`
```

> value - use it's value for display name in room list instead of global room name. In room page header (over timeline)

Reviewer 2: ```suggestion
attribute - use it's value for display name in room list instead of global room name.
```

> clients may show both names (global and personal) with explaining that this room have alternative personal name, that is

Reviewer 2: ```suggestion
Clients may show both names (global and personal) and explainin that this room has an alternative personal name, that is
```

seen only for current user.

## Adding, removing, renaming:

> Clients should provide the way to privately rename room and clearly explain the difference between global room name and

Reviewer 2: ```suggestion
Clients should provide a way to privately rename a room and clearly explain the difference between global room name and
```

personal name.

In room list this can be implemented via "Rename room personally" or "Set personal name for room" menu item in
right-click menu.

In room settings page personal name can be implemented via button "Set personal name for this room", which will be
available even if user have no permission to change the room name. If room personal name is filled, it should be shown
near the global room name, with ability to remove it.

# Server support

This MSC does not need any changes on server side.

# Potential issues

1. Users can set personal name for room identical as global name of room - this will give little negative effect: all
   further global room name changes will not change room name in client list. This can be solved via clearly describe
   the difference between global room name and personal name, and some warning that those names are identical with

>    ability to remove personal name.

Reviewer 2: I would reword this slightly. Something like.

1. Users can set a personal name that is similar or identical to the global room name. This may cause confusion as the client will not see future global name changes. Clients should consider providing the user a warning and explaining how personal and global room names work.

Author: Sorry for my English and thanks for all your suggestions! Hope I'm fixed all of them now ;-)

Reviewer 2: No worries. Thanks for the proposal 🙂
